lib/apiservers/engine/backends/container.go

```diff
@@ -925,7 +928,12 @@ func (c *Container) containerStart(name string, hostConfig *containertypes.HostC
 	if vc == nil {
 		return NotFoundError(name)
 	}
+	if !vc.TryLock(APITimeout) {
+		return ConcurrentAPIError(name, "ContainerStart")
```

Comment:
Is there no way to provide a useful message here?  i.e. the container is already being _fill in the blank_ -- essentially it's not a timeout, but something else is happening can't we tell the user what's happening?

Reply:
I'm not sure what you're saying.  We are supplying the API caller an error message.
```
func ConcurrentAPIError(name, request string) error {
	return derr.NewRequestConflictError(fmt.Errorf("A concurrent %s request is already in progress for container '%s'.  Please try again later.", request, name))
}
```
This is the error you will see if you're able to make overlapping ContainerStart request on the same container on a loaded system.  It is also what you see in the logs.

```
Dec 14 2017 01:56:40.373Z DEBUG op=281.2143: Container start failed due to error - A concurrent ContainerStart request is already in progress for container '9771b7879ec6c570a15a20c07cd8d6198da9d76e5eaff7231ecaefd8f4342b2b'.  Please try again later.
```

Reply:
You're right..I missed the fact that you were providing the request -- ie. `start`-- I originally read it as just a vanilla error when no request detail.